pkgs/development/libraries/agda/cubical-mini/default.nix

```diff
@@ -0,0 +1,40 @@
+{
+  lib,
+  mkDerivation,
+  fetchFromGitHub,
+  ghc,
+  cabal-install,
+}:
+
+mkDerivation rec {
+  pname = "cubical-mini";
+  version = "nightly-20241214";
+
+  src = fetchFromGitHub {
+    repo = pname;
+    owner = "cmcmA20";
+    rev = "ab18320018ddc0055db60d4bb5560d31909c5b78";
+    hash = "sha256-32qXY9KbProdPwqHxSkwO74Oqx65rTzoXtH2SpRB3OM=";
+  };
+
+  nativeBuildInputs = [
+    ghc
+    cabal-install
+  ];
+
+  buildPhase = ''
+    export HOME=$TMP
```

Comment:
`runHook preBuild` before and `runHook postBuild` after

Reply:
Done.

Reply:
The point of `preBuild` is generally to allow users to run commands before the `buildPhase`. If we use it ourselves, then if a user tries to override it the package will not build any more. We should just move these commands to `buildPhase`, between `runHook preBuild` and `make.

Reply:
Also maybe add a comment explaining why this dance is needed (the Makefile uses `cabal run ...`, and the default Cabal config makes it require an Internet connection).

Reply:
Changed and pushed.

Reply:
> Also maybe add a comment explaining why this dance is needed (the Makefile uses `cabal run ...`, and the default Cabal config makes it require an Internet connection).

Reply:
Sorry I didn't see your second comment the first time, because I didn't refresh the GitHub page. One moment.

Reply:
Added the comment too.